docs/modules/ROOT/pages/high_level/security_group_host_design.adoc

```diff
@@ -0,0 +1,583 @@
+We have added operation_type into each security group rule inside an SG configuration message.
+Therefore, only one SG rule is needed in the goal state message without the need to include 100s of existing rules when they have not changed.
+The DPM<->ACA schema (securitygroup.proto) already has that.
+
+[source,shell]
+------------------------------------------------------------
+    message SecurityGroupRule {
+        04 bytes - OperationType operation_type = 1;  <---------here
+        36 bytes - string security_group_id = 2;
+        36 bytes - string id = 3;
+        04 bytes - Direction direction = 4;
+        04 bytes - EtherType ethertype = 5;
+        04 bytes - Protocol protocol = 6;
+        04 bytes - uint32 port_range_min = 7;
+        04 bytes - uint32 port_range_max = 8;
+        18 bytes - string remote_ip_prefix = 9;
+        36 bytes - string remote_group_id = 10;
+    }
+------------------------------------------------------------
+
+The goal state message size reduction is significant.
+With Delta update, message size with one security group rule will be around 150 bytes plus outer goal state message and protobuf overhead (around 50 bytes, message total ~200 bytes).
+Without Delta update, full message size with 100 security group rules will be 150 bytes times 100 plus outer goal state message and protobuf overhead (around 50 bytes, message total ~15,050 bytes).
+
+=== Remote Security Group Rule Handling
+
+A Security Group rule can refer to a remote security group for both ingress and egress traffic.
+When a security group rule is added to security group A, customer can set security group B as the source (for ingress rule) or destination (for egress rule).
+This allows ports associated with security group B to access ports that are port of security group A, and vice versa.
+
+[source,shell]
+------------------------------------------------------------
++-----------+     +-----------+ 
+| Compute   |     | Compute   |
+| Node 1    |     | Node 2    |
+|           |     |           |
+|       OVS + <---| OVS       |
+|           |     |           |
+| Port 1    |     | Port 2    |
+| SG A      |     | SG B      |
++-----------+     +-----------+
+------------------------------------------------------------
+
+The default security group has rules that allow associated ports of the default rule to talk to each other.
+
+To support this using the minimal set of OpenFlow rules, we will mark remote ports with its associated security group using conjunctive flows discussed in the next session.
+We will update our neighbor configuration schema to include associated security group IDs.
+For L3 neighbor, we will leverage on-demand rules so that the OpenFlow rule for the marking is only setup when needed (for egress).
+For L2 neighbor, we will go ahead to setup its OpenFlow rules for the marking since it is limited by the subnet size.
```

Comment:
Line 166-168 is about L2 and L3 neighbors, not directly related to SG. Recommend to move them to other design doc, or create a sub-session for them.

Reply:
The design is the use L3/L2 neighbor state to support SG remote rule, therefore it is related to SG.